### ISSUE TITLE
[Enhancement] Optimize the performance of full screen feature

### ISSUE THREAD

**Issue by xiaogaozi** [victoralvesf/aonsoku#86]
I like the full screen feature, but the UI seems to become very lagging after switch to full screen.

https://github.com/user-attachments/assets/ff2008d1-8853-412b-8605-950219b86a47

**Comment by victoralvesf** [comment]
This is unexpected. The only known issue with this page is the queue list, which can become very laggy if it’s too long. However, a fix for that will be available soon.

The video shows the entire page lagging, which is new to me. I’ll need more details to investigate further.  

- Are you using the web version? If so, which browser?  
- Are you using the desktop app?
- What OS and architecture are you using?

**Comment by xiaogaozi** [comment]
> Are you using the web version? If so, which browser?

Yes, Chrome 131.0.6778.86.

> Are you using the desktop app?

No.

> What OS and architecture are you using?

macOS 15.1.1 and Intel chip (2019 MacBook Pro).

**Comment by victoralvesf** [comment]
@xiaogaozi I made some changes and was able to improve the performance slightly. I also added animations to enhance the experience.

Could you test this [deployment](https://aonsoku-git-feat-improve-e64583-victor-alves-projects-ccaa8240.vercel.app?_vercel_share=ul1FnchgtCt5Sqhh8cF39MPnbrpUXXnN) on your machine and verify the performance?

**Comment by kobayashi90** [comment]
@victoralvesf ive noticed an problem too, it automaticly switches the whole Web UI into Fullscreen.
Using the latest Aonsoku with Firefox 133.0 (64Bit) Windows

**Comment by victoralvesf** [comment]
> @victoralvesf ive noticed an problem too, it automaticly switches the whole Web UI into Fullscreen.
> 
> Using the latest Aonsoku with Firefox 133.0 (64Bit) Windows

This is the expected behavior (added on v0.6.0), in the future there will be a toggle in settings so users can enable or disable this feature.

**Comment by xiaogaozi** [comment]
@victoralvesf The performance is much better!

https://github.com/user-attachments/assets/c82ba154-f5e6-4db3-b230-e311ad73262e

**Comment by kobayashi90** [comment]
> This is the expected behavior (added on v0.6.0), in the future there will be a toggle in settings so users can enable or disable this feature.

I do hope it is soon, it breaks functionality on the WebUI.. thanks for letting me know tho.

**Comment by victoralvesf** [comment]
> > This is the expected behavior (added on v0.6.0), in the future there will be a toggle in settings so users can enable or disable this feature.
> 
> 
> 
> I do hope it is soon, it breaks functionality on the WebUI.. thanks for letting me know tho.

If you want the full screen experience, but doesn't want to stay on full screen, just press F11, then it will be back to normal web tab. Although, the settings page/dialog is planned to be launched soon.

**Comment by kobayashi90** [comment]
> > > This is the expected behavior (added on v0.6.0), in the future there will be a toggle in settings so users can enable or disable this feature.
> > 
> > 
> > I do hope it is soon, it breaks functionality on the WebUI.. thanks for letting me know tho.
> 
> If you want the full screen experience, but doesn't want to stay on full screen, just press F11, then it will be back to normal web tab. Although, the settings page/dialog is planned to be launched soon.

Are those settings possible to set via docker itself, so like enviorment variables?

**Comment by victoralvesf** [comment]
> Are those settings possible to set via docker itself, so like enviorment variables?

No, there aren't. This F11 trick is just a way to leave full screen on browsers.

**Comment by kobayashi90** [comment]
> No, there aren't. This F11 trick is just a way to leave full screen on browsers.

No what ive meant is if the settings page are things which can be set as docker variables in the future?

**Comment by victoralvesf** [comment]
> > No, there aren't. This F11 trick is just a way to leave full screen on browsers.
> 
> No what ive meant is if the settings page are things which can be set as docker variables in the future?

Some settings might be added to the configuration using environment variables, but in general, they are intended for the desktop app.

**Comment by victoralvesf** [comment]
This fix will be released in v0.7.0, which is planned to launch on December 9th.
